nixos/modules/services/databases/openldap.nix

```diff
@@ -24,20 +23,79 @@ let
       include ${cfg.rootpwFile}
     ''}
     directory ${cfg.dataDir}
-    ${cfg.extraDatabaseConfig}
+    ${if cfg.extraDatabaseConfig != null then cfg.extraDatabaseConfig else ""}
   '');
-  configOpts = if cfg.configDir == null then "-f ${configFile}"
-               else "-F ${cfg.configDir}";
-in
 
-{
-
-  ###### interface
+  configDir = if cfg.configDir != null then cfg.configDir else "/etc/openldap/slapd.d";
 
+  ldapValueType = let
+    singleLdapValueType = types.either types.str (types.submodule {
+      options = {
+        path = mkOption {
+          type = types.path;
+          description = ''
+            A path containing the LDAP attribute. This is included at run-time, so
+            is recommended for storing secrets.
+          '';
+        };
+      };
+    });
+  in types.either singleLdapValueType (types.listOf singleLdapValueType);
```

Comment:
You can change this to `types.coercedTo singleLdapValueType singleton (types.listOf singleLdapValueType)`, then any non-list definitions are automatically converted to a single-entry list.

Reply:
This does not work, I get the error: `error: value is a list while a string was expected, at /home/kai/Documents/code/nixpkgs/lib/attrsets.nix:222:5`.

I think it's to do with the fact that `coercedTo` doesn't work with submodules, even though the fact that wrapping in `types.either` stops the assertion from triggering:

```
      assert lib.assertMsg (coercedType.getSubModules == null)
        "coercedTo: coercedType must not have submodules (it’s a ${
          coercedType.description})";
```